File: examples/js/structs/src/person.ts
import { LHStruct, LHField } from 'littlehorse-client'
import { VariableType } from 'littlehorse-client/proto'
import { Address } from './address.js'

@LHStruct({ name: 'person' })
export class Person {
  @LHField(VariableType.STR)
  firstName!: string

  @LHField(VariableType.STR)
  lastName!: string

  @LHField({ struct: Address })
  homeAddress!: Address

  constructor(firstName?: string, lastName?: string, homeAddress?: Address) {
    if (firstName !== undefined) this.firstName = firstName
    if (lastName !== undefined) this.lastName = lastName
    if (homeAddress !== undefined) this.homeAddress = homeAddress
  }

  toString(): string {
    return `${this.firstName} ${this.lastName}`
  }
}

File: examples/go/structdef/structs/person.go
package structs

import (
	"fmt"

	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

const PersonStructDefName string = "person"
const PersonStructDefDescription string = "Person represents a person with a name and home address."

// Person represents a person with a name and home address.
type Person struct {
	FirstName   string  `json:"firstName"`
	LastName    string  `json:"lastName"`
	HomeAddress Address `json:"homeAddress" lh:",masked"`
}

func (Person) LHStructDef() littlehorse.LHStructDefInfo {
	return littlehorse.LHStructDefInfo{Name: PersonStructDefName, Description: PersonStructDefDescription}
}

func (p Person) String() string {
	return fmt.Sprintf("%s %s", p.FirstName, p.LastName)
}
